File: sdk/ruby/test/wire/releases_test.rb
# frozen_string_literal: true

require_relative "wiremock_test_case"

class ReleasesWireTest < WireMockTestCase
  def setup
    super

    @client = AgentTrunk::Client.new(
      access_token: "<token>",
      base_url: WIREMOCK_BASE_URL
    )
  end

  def test_releases_list_with_wiremock
    test_id = "releases.list.0"

    @client.releases.list(
      trunk_id: "trunkId",
      request_options: {
        additional_headers: {
          "X-Test-Id" => "releases.list.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "GET",
      url_path: "/v1/trunks/trunkId/promotion-requests",
      query_params: nil,
      expected: 1
    )
  end

  def test_releases_open_with_wiremock
    test_id = "releases.open.0"

    @client.releases.open(
      trunk_id: "trunkId",
      request_options: {
        additional_headers: {
          "X-Test-Id" => "releases.open.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "POST",
      url_path: "/v1/trunks/trunkId/promotion-requests",
      query_params: nil,
      expected: 1
    )
  end

  def test_releases_merge_with_wiremock
    test_id = "releases.merge.0"

    @client.releases.merge(
      trunk_id: "trunkId",
      promotion_id: "promotionId",
      request_options: {
        additional_headers: {
          "X-Test-Id" => "releases.merge.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "POST",
      url_path: "/v1/trunks/trunkId/promotion-requests/promotionId/merge",
      query_params: nil,
      expected: 1
    )
  end
end

File: sdk/rust/tests/releases_test.rs
use agenttrunk::prelude::*;

mod wire_test_utils;

#[tokio::test]
#[allow(unused_variables, unreachable_code)]
async fn test_releases_list_with_wiremock() {
    wire_test_utils::reset_wiremock_requests().await.unwrap();
    let wiremock_base_url = wire_test_utils::get_wiremock_base_url();

    let mut config = ClientConfig {
        token: Some("<token>".to_string()),
        ..Default::default()
    };
    config.base_url = wiremock_base_url.to_string();
    let client = AgentTrunk::new(config).expect("Failed to build client");

    let result = client
        .releases
        .list(
            &"trunkId".to_string(),
            &ReleasesListQueryRequest {
                ..Default::default()
            },
            None,
        )
        .await;

    assert!(result.is_ok(), "Client method call should succeed");

    wire_test_utils::verify_request_count("GET", "/v1/trunks/trunkId/promotion-requests", None, 1)
        .await
        .unwrap();
}

#[tokio::test]
#[allow(unused_variables, unreachable_code)]
async fn test_releases_open_with_wiremock() {
    wire_test_utils::reset_wiremock_requests().await.unwrap();
    let wiremock_base_url = wire_test_utils::get_wiremock_base_url();

    let mut config = ClientConfig {
        token: Some("<token>".to_string()),
        ..Default::default()
    };
    config.base_url = wiremock_base_url.to_string();
    let client = AgentTrunk::new(config).expect("Failed to build client");

    let result = client
        .releases
        .open(
            &"trunkId".to_string(),
            &OpenPromotionRequestInput {
                ..Default::default()
            },
            None,
        )
        .await;

    assert!(result.is_ok(), "Client method call should succeed");

    wire_test_utils::verify_request_count("POST", "/v1/trunks/trunkId/promotion-requests", None, 1)
        .await
        .unwrap();
}

#[tokio::test]
#[allow(unused_variables, unreachable_code)]
async fn test_releases_merge_with_wiremock() {
    wire_test_utils::reset_wiremock_requests().await.unwrap();
    let wiremock_base_url = wire_test_utils::get_wiremock_base_url();

    let mut config = ClientConfig {
        token: Some("<token>".to_string()),
        ..Default::default()
    };
    config.base_url = wiremock_base_url.to_string();
    let client = AgentTrunk::new(config).expect("Failed to build client");

    let result = client
        .releases
        .merge(&"trunkId".to_string(), &"promotionId".to_string(), None)
        .await;

    assert!(result.is_ok(), "Client method call should succeed");

    wire_test_utils::verify_request_count(
        "POST",
        "/v1/trunks/trunkId/promotion-requests/promotionId/merge",
        None,
        1,
    )
    .await
    .unwrap();
}
